dotnet/src/SemanticKernel.UnitTests/Planning/PlanTests.cs

```diff
@@ -338,17 +337,17 @@ public async Task StepExceptionIsThrownAsync()
         // Arrange
         var (kernel, serviceProvider, serviceSelector) = this.SetupKernel();
 
-        var mockFunction = new Mock<ISKFunction>();
-        mockFunction.Setup(x => x.InvokeAsync(It.IsAny<Kernel>(), It.IsAny<SKContext>(), null, It.IsAny<CancellationToken>()))
-            .Throws(new ArgumentException("Error message"));
-        mockFunction.Setup(x => x.GetMetadata()).Returns(() => new SKFunctionMetadata("functionName"));
+        var mockFunction = new KernelFunctionMock
+        {
+            InvokeCoreDelegate = (_, _, _, _) => throw new ArgumentException("Error message"),
```

Comment:
I would suggest using the 'SKFunctionFactory.CreateFromMethod' method instead of mocking the function, as it typically results in cleaner syntax: 'var function = SKFunctionFactory.CreateFromMethod(() => {throw new ...}, "function1");' and to use mocks only when the 'CreateFromMethod' method is insufficient or not applicable.

Reply:
Done